src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -255,25 +255,27 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		}
 		$user_cpt         = array();
 		$post_type_filter = 'wp_global_styles';
+		$stylesheet       = $theme->get_stylesheet();
 		$args             = array(
-			'numberposts' => 1,
-			'orderby'     => 'date',
-			'order'       => 'desc',
-			'post_type'   => $post_type_filter,
-			'post_status' => $post_status_filter,
-			'tax_query'   => array(
+			'posts_per_page'      => 1,
+			'orderby'             => 'post_date',
+			'order'               => 'desc',
+			'post_type'           => $post_type_filter,
+			'post_status'         => $post_status_filter,
+			'ignore_sticky_posts' => true,
+			'no_found_rows'       => true,
+			'tax_query'           => array(
 				array(
 					'taxonomy' => 'wp_theme',
 					'field'    => 'name',
-					'terms'    => $theme->get_stylesheet(),
+					'terms'    => $stylesheet,
 				),
 			),
 		);
 
 		$cache_key = sprintf( 'wp_global_styles_%s', md5( serialize( $args ) ) );
-		$post_id   = wp_cache_get( $cache_key );
-
-		if ( (int) $post_id > 0 ) {
+		$post_id   = (int) get_transient( $cache_key );
```

Comment:
Is there a `wp_cache_delete` call that needs to be updated to use the transient too? 

If the post is subsequently deleted then I think the cached value will fall out of date, so there probably should be.

Reply:
That is a good point. If the post is deleted in the database, then this will return null. Which doesn't seem right..

Reply:
Switching to a transient here results in a lot less queries per page.

Reply:
@draganescu `wp_cache_get` is useful if you are running object caching. If object caching is not enabling, which it is not in most cases, it is only cached in page memory. `get_transient` caches in the option table. So no matter what you are running, it results in less queries. As one query to options table is better than multiple queries that WP_Query generates.